Index: test-project/build-logic/groovy-build-logic/src/main/groovy/test/project/build/TestGroovyPlugin.groovy
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/test-project/build-logic/groovy-build-logic/src/main/groovy/test/project/build/TestGroovyPlugin.groovy b/test-project/build-logic/groovy-build-logic/src/main/groovy/test/project/build/TestGroovyPlugin.groovy
--- a/test-project/build-logic/groovy-build-logic/src/main/groovy/test/project/build/TestGroovyPlugin.groovy	(revision c88205e9ca573c625403202c42a9e8f64656e431)
+++ b/test-project/build-logic/groovy-build-logic/src/main/groovy/test/project/build/TestGroovyPlugin.groovy	(date 1652965995195)
@@ -12,11 +12,11 @@
     @Override
     void apply(Project project) {
         project.tasks.withType(ProcessResources).configureEach { ProcessResources it ->
-            it.destinationDir = project.layout.buildDirectory.dir("new-resources").get().asFile
+            it.destinationDir.set(project.layout.buildDirectory.dir("new-resources"))
         }

         project.tasks.withType(GroovyCompile).configureEach { GroovyCompile it ->
-            it.options.compilerArgs << "-proc:none"
+            it.options.compilerArgs.add("-proc:none")
         }
     }
 }
Index: test-project/build-logic/java-build-logic/src/main/java/test/project/build/TestJavaPlugin.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/test-project/build-logic/java-build-logic/src/main/java/test/project/build/TestJavaPlugin.java b/test-project/build-logic/java-build-logic/src/main/java/test/project/build/TestJavaPlugin.java
--- a/test-project/build-logic/java-build-logic/src/main/java/test/project/build/TestJavaPlugin.java	(revision c88205e9ca573c625403202c42a9e8f64656e431)
+++ b/test-project/build-logic/java-build-logic/src/main/java/test/project/build/TestJavaPlugin.java	(date 1652965995184)
@@ -16,8 +16,8 @@
         project.getPluginManager().apply("java-library");

         project.getTasks().withType(JavaCompile.class).configureEach(compile -> {
-            compile.setSourceCompatibility("11");
-            compile.setTargetCompatibility("11");
+            compile.getSourceCompatibility().set("11");
+            compile.getTargetCompatibility().set("11");
             compile.getOptions().getCompilerArgs().add("-proc:none");
             compile.doLast(new Action<Task>() {
                 @Override
@@ -27,7 +27,7 @@
             });
         });
         project.getTasks().withType(ProcessResources.class).configureEach(processResources -> {
-            processResources.setDestinationDir(project.getLayout().getBuildDirectory().dir("new-resources").get().getAsFile());
+            processResources.getDestinationDir().set(project.getLayout().getBuildDirectory().dir("new-resources"));
         });

         project.getDependencies().constraints(constraints -> {
Index: test-project/build-logic/groovy-build-logic/src/main/groovy/test.project.build.groovy-application-conventions.gradle
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/test-project/build-logic/groovy-build-logic/src/main/groovy/test.project.build.groovy-application-conventions.gradle b/test-project/build-logic/groovy-build-logic/src/main/groovy/test.project.build.groovy-application-conventions.gradle
--- a/test-project/build-logic/groovy-build-logic/src/main/groovy/test.project.build.groovy-application-conventions.gradle	(revision c88205e9ca573c625403202c42a9e8f64656e431)
+++ b/test-project/build-logic/groovy-build-logic/src/main/groovy/test.project.build.groovy-application-conventions.gradle	(date 1652965995179)
@@ -8,7 +8,7 @@
 tasks.withType(AbstractCompile).configureEach {
     targetCompatibility = "11"
     sourceCompatibility = "11"
-    classpath += files()
+    classpath.from(files())
     doLast {
         println("Hello from Groovy plugin, classpath length: ${classpath.files.size()}")
     }
Index: test-project/build-logic/kotlin-build-logic/src/main/kotlin/test/project/build/TestKotlinPlugin.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/test-project/build-logic/kotlin-build-logic/src/main/kotlin/test/project/build/TestKotlinPlugin.kt b/test-project/build-logic/kotlin-build-logic/src/main/kotlin/test/project/build/TestKotlinPlugin.kt
--- a/test-project/build-logic/kotlin-build-logic/src/main/kotlin/test/project/build/TestKotlinPlugin.kt	(revision c88205e9ca573c625403202c42a9e8f64656e431)
+++ b/test-project/build-logic/kotlin-build-logic/src/main/kotlin/test/project/build/TestKotlinPlugin.kt	(date 1652965995189)
@@ -9,7 +9,7 @@
 class TestKotlinPlugin : Plugin<Project> {
     override fun apply(project: Project) {
         project.tasks.withType<ProcessResources>().configureEach {
-            destinationDir = project.layout.buildDirectory.dir("new-resources").get().asFile
+            destinationDir = project.layout.buildDirectory.dir("new-resources")
         }

         project.tasks.withType<JavaCompile>().configureEach {
